kubernetes/outreach.libsonnet

```diff
@@ -225,23 +225,24 @@ k + kubecfg {
 
     metadata+: {
       annotations+: {
-        # ALB ANNOTATIONS
+        // ALB ANNOTATIONS
         'kubernetes.io/ingress.class': 'alb',
         'alb.ingress.kubernetes.io/ssl-redirect': '443',
-        'alb.ingress.kubernetes.io/group.name': groupName, // IngressGroup feature enables you to group multiple Ingress resources together and use a single ALB
-        'alb.ingress.kubernetes.io/tags': 'cost=ingress_alb,outreach:environment=%s,kubernetesCluster=%s' % [cluster.environment, cluster.fqdn], 
+        'alb.ingress.kubernetes.io/group.name': groupName,  // IngressGroup feature enables you to group multiple Ingress resources together and use a single ALB
+        'alb.ingress.kubernetes.io/tags': 'cost=ingress_alb,outreach:environment=%s,kubernetesCluster=%s' % [cluster.environment, cluster.fqdn],
         'alb.ingress.kubernetes.io/listen-ports': '[{"HTTP":80},{"HTTPS":443}]',
-        'alb.ingress.kubernetes.io/actions.ssl-redirect': '{"Type": "redirect", "RedirectConfig": { "Protocol": "HTTPS", "Port": "443", "StatusCode": "HTTP_301"}}', // Redirect http to https
+        'alb.ingress.kubernetes.io/actions.ssl-redirect': '{"Type": "redirect", "RedirectConfig": { "Protocol": "HTTPS", "Port": "443", "StatusCode": "HTTP_301"}}',  // Redirect http to https
         'alb.ingress.kubernetes.io/ssl-policy': 'ELBSecurityPolicy-TLS13-1-2-Res-2021-06',
         'alb.ingress.kubernetes.io/scheme': scheme,
-        'alb.ingress.kubernetes.io/load-balancer-attributes': 'routing.http.drop_invalid_header_fields.enabled=true,access_logs.s3.enabled=true,access_logs.s3.bucket=outreach-aws-lb-controller-logs-%s,access_logs.s3.prefix=%s,idle_timeout.timeout_seconds=%s' % [cluster.region, groupName, idleTimeoutSeconds], 
+        'alb.ingress.kubernetes.io/load-balancer-attributes': 'routing.http.drop_invalid_header_fields.enabled=true,access_logs.s3.enabled=true,access_logs.s3.bucket=outreach-aws-lb-controller-logs-%s,access_logs.s3.prefix=%s,idle_timeout.timeout_seconds=%s' % [cluster.region, groupName, idleTimeoutSeconds],
         'alb.ingress.kubernetes.io/success-codes': '200-399',
+        'alb.ingress.kubernetes.io/target-type': 'ip',
```

Comment:
Hello @kaldorn , based on previous incident caused by changes in jsonnet-libs [ICM-676](https://outreach-io.atlassian.net/browse/ICM-676), I need to ask for a condition to ensure this is not spreading everywhere without a control. One of the learnings in the incident was that we need to ensure changes are deployed on staging where we need a chance to see how this works and whether it affects anything. Could you please add such condition ? 

Also, I am missing some context in the PR. Why are we doing this change ? What is the benefit of doing this, and is there any downside of this ? Is this part of any bigger project (like the CrossAZ networking which @jan-ludvik is leading) ?

I am blocking the PR until we have those questions answered and staging condition in place.

Reply:
discussion moved to slack 👉 https://outreach-hq.slack.com/archives/C06HLD1DARH/p1707388405057919